dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -191,6 +192,7 @@ DawnCompiler::lowerToIIR(std::shared_ptr<SIR> const& stencilIR) {
     optimizer.pushBackPass<PassSetSyncStage>();
     // validation checks after parallelisation
     optimizer.pushBackPass<PassLocalVarType>();
+    optimizer.pushBackPass<PassRemoveScalars>();
```

Comment:
For now this pass is not required to run. It does not contribute to validity, right? This will only come in if we have the stage-splitter / stage location type

Reply:
Correct. Still worth keeping it for integration tests (e.g. checking outputs are numerically the same even after removal of scalars).

Reply:
That is not how this works. 

Create a group for this pass

Reply:
But `PassLocalVarType` is also in the parallel group, what is the difference? Both will be necessary to have a valid parallel IIR, which is currently not valid because the stage location types are incorrect.
I'd say it makes sense to test them in the group where they will be at the end, not in another one which will be removed. Unless you suggest to create a group that runs before the parallel group.

Reply:
No, I'm not saying that. When we discussed where to put `PassLocalVarType` you made me notice that, at the moment, it's not required for stage splitting so it doesn't need to be put before `PassStageSplitter`, so we put it at the end of the group. I'd say the same goes for `PassRemoveScalars`.
As we cannot implement everything in one big step (flag variables + remove scalars + fix stage location type), it seems reasonable to incrementally add passes to the group in order to get them covered by integration tests.

Reply:
sorry. I still do not follow.

In the parallel group there should only be the passes that are *absolutely* required to generate valid, parallel IR.
So if both of these passes are only required for optimization they should appear only in a group below (or be their own group) and if they are necessary up here, they can stay

Reply:
They are not an optimization, they are required by `PassStageSplitter`. But currently `PassStageSplitter` does not the correct thing, as it does not set the correct location types of stages and thus it does not use the output of these passes.

Reply:
ok